packages/library-sync/src/commonMain/kotlin/io/realm/mongodb/AppConfiguration.kt

```diff
@@ -16,37 +16,74 @@
 
 package io.realm.mongodb
 
+import io.realm.internal.platform.singleThreadDispatcher
 import io.realm.mongodb.internal.AppConfigurationImpl
 import kotlinx.coroutines.CoroutineDispatcher
 
 /**
- * TODO
+ * An **AppConfiguration** is used to setup linkage to a MongoDB Realm application.
+ *
+ * Instances of a AppConfiguration can only created by using the
+ * {@link AppConfiguration.Builder} and calling its
+ * {@link AppConfiguration.Builder#build()} method.
+ * <p>
+ * Configuring a App is only required if the default settings are not enough. Otherwise calling
+ * {@code App("app-id")} is sufficient.
  */
 interface AppConfiguration {
+
     val appId: String
+    // TODO Consider replacing with URL type, but didn't want to include io.ktor.http.Url as it
+    //  requires ktor as api dependency
     val baseUrl: String
     val networkTransportDispatcher: CoroutineDispatcher
 
     companion object {
         /**
-         * TODO
+         * The default url for MongoDB Realm applications.
+         *
+         * @see Builder#baseUrl(String)
          */
         const val DEFAULT_BASE_URL = "https://realm.mongodb.com"
 
         /**
-         * TODO
+         * The default header name used to carry authorization data when making network requests
+         * towards MongoDB Realm.
          */
         const val DEFAULT_AUTHORIZATION_HEADER_NAME = "Authorization"
     }
-}
 
-// TODO Create full blown builder for this
-// FIXME Initialize with proper multithreaded dispatcher
-//  https://github.com/realm/realm-kotlin/issues/450
-fun appConfigurationOf(
-    appId: String,
-    baseUrl: String,
-    dispatcher: CoroutineDispatcher
-): AppConfiguration {
-    return AppConfigurationImpl(appId, baseUrl, dispatcher)
+    /**
+     * Builder used to construct instances of a {@link AppConfiguration} in a fluent manner.
```

Comment:
The format you used is Javadoc, not dokka.
```suggestion
     * Builder used to construct instances of an [AppConfiguration] in a fluent manner.
```